assets/js/modules/analytics/datastore/settings.test.js

```diff
@@ -784,6 +948,38 @@ describe( 'modules/analytics settings', () => {
 				).toThrow( INVARIANT_INVALID_PROFILE_NAME );
 			} );
 
+			it( 'does not require a valid profile name when the `ga4Reporting` feature flag is enabled', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.receiveGetExistingTag( null );
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.setSettings( validSettings );
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.setProfileID( PROFILE_CREATE );
+
+				// Ensure the validation is not triggered when the profile name is not set at all.
+				expect( () =>
+					registry
+						.select( MODULES_ANALYTICS )
+						.__dangerousCanSubmitChanges()
+				).not.toThrow( INVARIANT_INVALID_PROFILE_NAME );
+
+				// Set an invalid/empty profile name.
+				registry
+					.dispatch( CORE_FORMS )
+					.setValues( FORM_SETUP, { profileName: '' } );
+
+				// Ensure the validation is not triggered when an empty profile name is set.
+				expect( () =>
+					registry
+						.select( MODULES_ANALYTICS )
+						.__dangerousCanSubmitChanges()
+				).not.toThrow( INVARIANT_INVALID_PROFILE_NAME );
```

Comment:
We should be using `__dangerousCanSubmitChanges` to make assertions about why changes cannot be submitted, otherwise it isn't particularly useful that an error was not thrown because a different error thrown (i.e. due to an earlier failed validation) will pass the assertion here but not actually ensure that the specific validation passes.

For this reason, we should be expecting `canSubmitChanges` is `true` similar to the above to ensure all validation passes.

Reply:
Gotcha. That makes sense 👍